Fix #690: Create README.md

ma8ma opened the pull request:
This PR fixes #690, see [here](https://github.com/ma8ma/vimperator-labs/blob/create-readme.md/README.md) for looks.

README.md for people who would like to hack vimperator, it shows the
latest change and the method of building own version. Note that
contributor is required to edit the What's New topic.

Note: Issues or Pull requests numbers don't automatically make links to
the repository, but just leave it as it is because reduce contributor's
burden.

timss commented:
Awesome! I actually started spec'ing the readme myself right before you
made #690, but this is pretty much what I envisioned anyway :)

I'll look over my notes later today and see if there's anything more that I
feel should be added before merging this.

Thanks.

On Feb 5, 2017 13:14, "Masayuki Yamamoto" <notifications@github.com> wrote:

This PR fixes #690
<https://github.com/vimperator/vimperator-labs/issues/690>.
README.md for people who would like to hack vimperator, it shows the
latest change and the method of building own version. Note that
contributor is required to edit the What's New topic.

Note: Issues or Pull requests numbers don't automatically make links to
the repository, but just leave it as it is because reduce contributor's
burden.
------------------------------
You can view, comment on, or merge this pull request online at:

  https://github.com/vimperator/vimperator-labs/pull/694
Commit Summary

   - Fix #690: Create README.md

File Changes

   - *M* CONTRIBUTING.md
   <https://github.com/vimperator/vimperator-labs/pull/694/files#diff-0>
   (2)
   - *A* README.md
   <https://github.com/vimperator/vimperator-labs/pull/694/files#diff-1>
   (171)
   - *M* vimperator/NEWS
   <https://github.com/vimperator/vimperator-labs/pull/694/files#diff-2>
   (1)

Patch Links:

   - https://github.com/vimperator/vimperator-labs/pull/694.patch
   - https://github.com/vimperator/vimperator-labs/pull/694.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/vimperator/vimperator-labs/pull/694>, or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABqqeg25f5q-tcRCXPssKNXR6syO9pq_ks5rZb0VgaJpZM4L3er6>
.

Segaja commented:
Maybe it makes sense to then remove the NEWS file, because it seems the information in there and in the README are redundant. Furthermore I would link to the CONTRIBUTING file from within the README.

gkatsev commented:
I would prefer to move the news file into a CHANGELOG.md rather than having it in the README.

samrocketman commented:
Perhaps a small snippet explaing to new visitors: when the AMO version and GH release differ it's because the release is going through the AMO review process.

Also, I would discourage using acronyms without explaining their meaning.  Example,
> `addons.mozillla.org` (AMO)
is sufficient.  It's good to fully write the acronym and then abbreviate it for later occurrences.

allanwind commented:
Would it be possible to, at least temporarily, update the website with a pointer to github?  I was just about to switch to palemoon just to have access to vimperator.  Also, the mozilla wiki  kinda buries the key you need to set in the middle.  Maybe just reference it inline, so at least, you know what to search for?

timss commented on README.md:
```diff
@@ -0,0 +1,171 @@
+Vimerator-labs
```
Small typo :)

ma8ma replied:
Oops, that's terrible my mistake! :scream:

timss commented on README.md:
```diff
@@ -0,0 +1,171 @@
+Vimerator-labs
+==============
+
+![amo release][amo_version] ![github release][github_version]
```
Would be cool if these also linked to the AMO and GitHub release pages, like so:

    [![amo release][amo_version]][amo_release] [![github release][github_version]][github_release]

    [amo_version]: https://img.shields.io/amo/v/vimperator.svg
    [amo_release]: https://addons.mozilla.org/en-US/firefox/addon/vimperator/
    [github_version]: https://img.shields.io/github/release/vimperator/vimperator-labs.svg
    [github_release]: https://github.com/vimperator/vimperator-labs/releases/latest

timss commented on README.md:
```diff
@@ -0,0 +1,171 @@
+
+- Download add-on from AMO (signed)
+
+  > Note: If AMO version is previous than repo's, you can build own version
```
Good idea to mention this, but just some small language improvements:

>Note that if the version on [AMO](https://addons.mozilla.org/en-US/firefox/addon/vimperator/) is older than the [latest release](https://github.com/vimperator/vimperator-labs/releases/latest), you can build your own version and use it until AMO release is updated.

Also, might want to re-use these URLs from the shields.io section.

timss commented on README.md:
```diff
@@ -0,0 +1,171 @@
+  > Note: Since version 48, Installing add-on to Firefox has required signing.
+  > Unsigned add-on is for Unbranded Builds. See [mozilla wiki][wiki_signing].
+
+  1. Enter [Releases page](https://github.com/vimperator/vimperator-labs/releases).
```
Could use the `https://github.com/vimperator/vimperator-labs/releases/latest` URL to link directly to the latest release.

timss commented on README.md:
```diff
@@ -0,0 +1,171 @@
+
+[wiki_signing]: https://wiki.mozilla.org/Add-ons/Extension_Signing
+
+Build own version
```
Hm, this might be a tad too much to include in the readme. The instructions are good (:+1:), but might fit better in the wiki. Opinion @gkatsev ?

gkatsev replied:
nobody reads wikis :P

timss replied:
Not if they're not linked to!

ma8ma commented:
Thank you folks to review PR! 😄 I updated README, split a commit to enable revert in a hurry (marked WIP).
@Segaja @gkatsev What's New topic won't show old change (it's only pre-release and latest release), so I like CHANGELOG.md as storeroom for old change.
@timss I'm not sure whether better to move `Build own version` topic into wiki, but I agree with you, it's more details than others.

gkatsev commented:
I don't think that having the "what's new" section in the README is worth the extra maintenance of moving stuff from the README to the CHANGELOG.md or even having both together. Also, if you have a CHANGELOG.md, if we update how we merge things, we can even have the changelog get generator automatically when we cut a release (https://github.com/conventional-changelog/conventional-changelog-cli)

samrocketman commented:
Also, glad to see you decided to format 80 chars wide.  I regularly use vim to develop so typically look at READMEs in the terminal.

Quick `vim` tip: You can select whole paragraphs of test (`shit+V` to select a whole line).  Once you have selected the paragraph, then in normal mode type `g` and `q`.  The paragraph will auto-format to 80 characters.  In `vim`, see also `:help gq`.

SammysHP commented:
Two remarks:

#### Changelog

I think that the README should not include a changelog. It should explain what Vimperator does and how you can use it, maybe how you can extend and build it (if not covered in a separate development/contribute file).

#### Build process

Have you actually tested the signing method that you describe? I doubt that it will work. AFAIK (and that's what I wrote in private.properties) it must be done like

1. Register on AMO
1. Copy and edit private.properties.in
1. Build via `make xpi` (sign cannot work because the addon wasn't configured on AMO yet)
1. Create new addon on AMO and set it to unlisted (!), upload xpi etc.
1. Now you can run `make sign` to build, upload the unsigned xpi and download the signed one.

Ghosy commented:
Line 148 of `README.md` seems to be a bit off.

Issue: "Hove you ever submit own version to AMO?"

ma8ma commented:
I updated README.md, the What's New topic is gone. It will be a new issue.
@gkatsev I have an interest in auto generation for the changelog. Although it needs a consensus of current contributors, I'd agree to change message format.
@SammysHP As you pointed out, I had been confused for first signing method. I hope that it will fit by this modification.

samrocketman commented:
What's the status of this README?  I would rather see it merged and additional changes proposed later.  No README is detrimental to any project.

I use vimperator all the time but didn't give it a second look when I saw it didn't have a README.  Considering there's a lot of open source projects competing for developers, not having a README will make or break participation.

timss commented:
Was waiting to see if there was any additional feedback from others who had commented, but I agree with @samrocketman that it's better to let it be merged and if anyone feels like anything should be changed, added or whatever that can be done after a merge :)

Thanks again for all your work and updating the readme based on feedback! :+1:

samrocketman commented:
Thanks @ma8ma.  Documentation is not always easy to write.  I appreciate you taking the time.
